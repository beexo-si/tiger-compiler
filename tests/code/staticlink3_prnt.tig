let
    function f(n:int) : int =
        let
            var t := n+1
            function g(x:int):int =
                let
                    function h(y:int):int = (print("En h\n"); y+x+t)
                in
                    print("En g\n"); h(3)
                end
	in
            print("En f\n"); g(n+1)
        end
in
    f(2)
end
/* Caso 2 de static link: nivel_llamada > nivel_llamante */
