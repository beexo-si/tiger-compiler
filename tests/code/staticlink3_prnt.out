bien!
yes!!
En f
En g
En h
Return code: 9
